solana-programs/anchor/audius-data/programs/audius-data/src/lib.rs

```diff
@@ -170,8 +176,18 @@ pub mod audius_data {
         Ok(())
     }
 
+    /// Permissioned function to log an update to Admin metadata
+    pub fn update_admin(ctx: Context<UpdateAdmin>,  is_write_enabled: bool) -> Result<()> {
+        msg!("Audius::UpdateAdmin");
+        if ctx.accounts.admin.authority != ctx.accounts.admin_authority.key() { // could be has_one
```

Comment:
i think these checks could be more standardized with anchor like using has_one. looking into that now.